syntax="proto3";

package cpp_chat;

message Empty {}

message Message {
  string user_id = 1;
  string text = 2;
}

message ChatRoom {
  string room_name = 1;
  User user = 2;
}

message User {
  string user_id = 1;
}

message SimpleResponse {
  bool responseFlag = 1;
}

service chat {
  rpc login(User) returns (SimpleResponse) {};

  rpc reLogin(stream User) returns (SimpleResponse) {};

  rpc createChatRoom(ChatRoom) returns (SimpleResponse) {};

  rpc chatRoomList(Empty) returns (stream ChatRoom) {};

  rpc enterChatRoom(ChatRoom) returns (SimpleResponse) {};

  rpc chatting(stream Message) returns (stream Message) {};
}
